src/model/base.ts

```diff
@@ -1,12 +1,82 @@
-import { Tensor, tensor, RecursiveArray } from '@tensorflow/tfjs-core';
+import { Tensor, RecursiveArray, tensor, tidy, dispose } from '@tensorflow/tfjs-core';
 import { checkArray } from '../utils/validation';
 import { OneHotEncoder } from '../preprocess';
 import { OneHotDropTypes } from '../preprocess/encoder';
 
+export type FeatureInputType = Tensor | RecursiveArray<number>;
+export type LabelInputType = Tensor | RecursiveArray<number>;
+
 export type ClassMap = {
   [ key: string ]: number
 }
 
+export abstract class BaseEstimator {
+  public estimatorType: string;
+  public nFeature: number;
+
+  /**
+   * Check if input feature match the required feature size. If reset is true,
+   * reset nFeature = x.shape[1]
+   *
+   * @param x Input feature of shape (nSample, nFeatures).
```

Comment:
```suggestion
   * @param input feature of shape (nSample, nFeatures).
```

In tsdoc, we don't need to specify the variable name.